subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/junit/JUnitTestClassExecuter.java

```diff
@@ -80,12 +80,21 @@ private void runTestClass(String testClassName) throws ClassNotFoundException {
             ));
         }
 
+        Request request = Request.aClass(testClass);
+        Runner runner = request.getRunner();
+
         if (!options.getIncludedTests().isEmpty()) {
-            filters.add(new MethodNameFilter(options.getIncludedTests()));
+            TestSelectionMatcher matcher = new TestSelectionMatcher(options.getIncludedTests());
+
+            // If the test class name matches, skip the filter. This may be a test suite
+            // or other parent of multiple tests, in which case (if the class name matches)
+            // we should run the entire suite without requiring each individual test name
+            // to be explicitly included.
+            if (runner.getDescription().isSuite() && !matcher.matchesTest(testClassName, null)) {
```

Comment:
What if some runner returns `isSuite == false`? Looks like we would just execute it, even if the name does not match. We need a test for that.